test/functorch/test_eager_transforms.py

```diff
@@ -3508,6 +3510,123 @@ def forward(self, x_1):
     """)
 
 
+def construct_sum_pyop():
+    mysum = PyOperator("mysum")
+
+    @mysum.py_impl(torch._C._functorch.TransformType.Vmap)
+    def mysum_batch_rule(interpreter, x, dim):
+        if not torch._C._functorch.is_batchedtensor(x):
+            with interpreter.lower():
+                x = x.view_as(x)  # unnecessary, just here to test the dispatch
+                return mysum(x, dim)
+
+        bdim = torch._C._functorch.maybe_get_bdim(x)
+        value = torch._C._functorch.get_unwrapped(x)
+
+        with interpreter.lower():
+            value = value.movedim(bdim, 0)
+            result = mysum(value, dim + 1)
+
+        return torch._C._functorch._add_batch_dim(result, 0, interpreter.level())
+
+    @mysum.py_impl(torch._C._functorch.TransformType.Grad)
+    def mysum_grad_rule(interpreter, x, dim):
+        level = interpreter.level()
+
+        class MySum(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, x, dim):
+                ctx.x_shape = x.shape
+                ctx.dim = dim
+                x = torch._C._functorch._unwrap_for_grad(x, level)
```

Comment:
Maybe it's not super important to this PR, but would be cool to see an example where you save for backward since I guess there is the question of whether you want to save the wrapped or unwrapped tensors.

Reply:
Wrapped tensors should be saved for backward, because that's what functorch does today for regular C++ operators. There's a separate question of *if* wrapped tensors should be saved for backward. The answer to that is probably "no", but this is how the functorch design has been since the prototype

Reply:
Oh right, that makes sense. Making it so that tensors can be unwrapped before saving for backward would probably require changes to the saved variable logic, because otherwise the metadata of the inner tensor would be cleared too.